### ISSUE TITLE
`no such file or directory, lstat` between `analyze()` calls if directory changes

### ISSUE THREAD

**Issue by kachkaev** [Nixinova/Linguist#4]
👋 again @Nixinova 😅

Here is another small bug 🙂 As I’ve mentioned in #3, I work with two folders: `my-analytics-tool` and `my-analytics-tool-data`. The second folder contains the repo I want to analyse.

To create a time series with historic changes in the number of files, I do the following:

1. Clone a repo into `my-analytics-tool-data`
1. List commits in `my-analytics-tool-data`
1. Checkout commit `x`
1. Run linguist
1. Save a JSON summary for commit `x`
1. Pick another commit and continue from step 3.

So I’ve got a single script which calls `analyze("/path/to/my-analytics-tool-data")` multiple times. The contents of the analysed folder changes between the calls. When this happens, the function throws with the following message:

```txt
Error: ENOENT: no such file or directory, lstat '/path/to/my-analytics-tool-data/path/to/deleted/file.ext'
    at Object.lstatSync (fs.js:1115:3)
    at Object.analyse (/path/to/my-analytics-tool/node_modules/linguist-js/dist/index.js:106:26)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  errno: -2,
  syscall: 'lstat',
  code: 'ENOENT',
  path: '/path/to/my-analytics-tool-data/path/to/deleted/file.ext'
}
```

It’d be great to have an option to reset cache before running analyze or just make it default behaviour.

Regardless of this small glitch, thanks a lot for your effort! I’ve been waiting for this port for years 😁 👍👍👍👍

**Comment by Nixinova** [comment]
I'll see if I can trap that error, thanks for reporting, and glad to be of help with this port.

**Comment by Nixinova** [comment]
Can't seem to reproduce in v2.0.2 even though I didn't touch any relevant code.

**Comment by Nixinova** [comment]
Should be fixed in v2.0.2.

[Repro folder](https://github.com/Nixinova/Linguist/files/7133775/linguist.4.zip)

**Comment by kachkaev** [comment]
No more crashing in v2.0.2 🎉
